src/framework/filterControl/forms/DateRangeFilterForm.jsx

```diff
@@ -0,0 +1,108 @@
+
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import { DateRange as ReactDateRange } from 'react-date-range';
+import React, {
+  Component,
+} from 'react';
+
+import {
+  Filter,
+  FilterOption,
+} from '../../services';
+
+import PrimaryButton from '../../button/PrimaryButton.jsx';
+
+export default class DateRangeFilterForm extends Component {
+
+  constructor(props) {
+    super(props);
+
+    this.onChange = this.onChange.bind(this);
+    this.onClickAddButton = this.onClickAddButton.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+
+    let dateRange;
+    if (this.props.comparisonValue === undefined) {
+      dateRange = {
+        startDate: moment(new Date()),
+        endDate: moment(new Date()),
+      };
+    } else {
+      dateRange = {
+        startDate: moment(this.props.comparisonValue.startDate),
+        endDate: moment(this.props.comparisonValue.endDate),
+      };
+    }
+
+    this.state = {
+      dateRange,
+    };
+  }
+
+  onClickAddButton() {
+    const dateRange = {
+      startDate: this.state.dateRange.startDate.toDate(),
+      endDate: this.state.dateRange.endDate.toDate(),
+    };
+    const filter = new Filter(
+      this.props.filterOption,
+      dateRange
+    );
+    this.props.onAddFilter(filter);
+  }
+
+  onKeyUp(event) {
```

Comment:
Do we really need this?

Also it's not bound to an event emitter, so the function is not getting called either way.